File: data-structure-algo/unique_charaters/unique.js
characters = "theawesomeness"

console.log(myFunction(characters))

function myFunction(characters) {
    let mySet = new Set()

    for (i = 0; i < characters.length; i++) {
        if(mySet.size == 0) {
            mySet.add(characters.charAt[i])
        } else if (mySet.has(characters.charAt[i])) {
            return false
        }
    }
    return true
}

console.log(myFunction_v2(characters))

function myFunction_v2(characters) {

    let mySet = new Set(characters)

    return characters.length == mySet.length
}

File: data-structure-algo/Arrays/unique_charaters/java/Unique.java
import java.util.*; 

class Unique {
    public static void main(String[] args) {
        
        Set<String> my_set = new HashSet<String>(); 

        for(int i = 0; i < args[0].length(); i++) {
            char c = args[0].charAt(i);
            String s = String.valueOf(c);
            my_set.add(s);
        }

        if (my_set.size() != args[0].length()) {            
            System.out.println("NOT");
        } else {
            System.out.println("The characters are UNIQUE");
        }
    }

}

